multiwallet.go

```diff
@@ -0,0 +1,428 @@
+
+	return mw, nil
+}
+
+func (mw *MultiWallet) Shutdown() {
+	log.Info("Shutting down dcrlibwallet")
+
+	// Trigger shuttingDown signal to cancel all contexts created with `contextWithShutdownCancel`.
+	mw.shuttingDown <- true
+
+	mw.CancelSync()
+
+	for _, w := range mw.wallets {
+		w.Shutdown()
+	}
+
+	if logRotator != nil {
+		log.Info("Shutting down log rotator")
+		logRotator.Close()
+	}
+
+	if mw.db != nil {
+		err := mw.db.Close()
+		if err != nil {
+			log.Errorf("db closed with error: %v", err)
+		} else {
+			log.Info("db closed successfully")
+		}
+	}
+}
+
+func (mw *MultiWallet) loadWallets() (int, error) {
+	query := mw.db.Select(q.True()).OrderBy("ID")
+	var wallets []Wallet
+	err := query.Find(&wallets)
+	if err != nil && err != storm.ErrNotFound {
+		return 0, err
+	}
+
+	mw.wallets = make(map[int]*LibWallet)
+	for _, w := range wallets {
+		w.walletDbDriver = mw.dbDriver
+		w.netType = mw.activeNet.Name
+		libWallet, err := NewLibWallet(&w)
+		if err != nil {
+			return 0, err
+		}
+
+		mw.wallets[w.ID] = libWallet
+	}
+
+	return len(wallets), nil
+}
+
+func (mw *MultiWallet) GetBackupsNeeded() int32 {
+	var backupsNeeded int32
+	for _, w := range mw.wallets {
+		if w.WalletOpened() && w.wallet.Seed != "" {
+			backupsNeeded++
+		}
+	}
+
+	return backupsNeeded
+}
+
+func (mw *MultiWallet) LoadedWalletsCount() int32 {
+	return int32(len(mw.wallets))
+}
+
+func (mw *MultiWallet) OpenedWalletsRaw() []int {
+	wallets := make([]int, 0)
+	for _, w := range mw.wallets {
+		if w.WalletOpened() {
+			wallets = append(wallets, w.wallet.ID)
+		}
+	}
+
+	return wallets
+}
+
+func (mw *MultiWallet) OpenedWallets() string {
+	wallets := mw.OpenedWalletsRaw()
+	jsonEncoded, _ := json.Marshal(&wallets)
+
+	return string(jsonEncoded)
+}
+
+func (mw *MultiWallet) OpenedWalletsCount() int32 {
+	return int32(len(mw.OpenedWalletsRaw()))
+}
+
+func (mw *MultiWallet) SyncedWalletCount() int32 {
+	var syncedWallet int32
+	for _, w := range mw.wallets {
+		if w.WalletOpened() && w.synced {
+			syncedWallet++
+		}
+	}
+
+	return syncedWallet
+}
+
+func (mw *MultiWallet) CreateNewWallet(privatePassphrase string, spendingPassphraseType int32) (*LibWallet, error) {
+
+	if mw.activeSyncData != nil {
+		return nil, errors.New(ErrSyncAlreadyInProgress)
+	}
+
+	seed, err := GenerateSeed()
+	if err != nil {
+		return nil, err
+	}
+
+	properties := &Wallet{
+		Seed:                   seed,
+		SpendingPassphraseType: spendingPassphraseType,
+		DiscoveredAccounts:     true,
+	}
+
+	return mw.createWallet(properties, seed, privatePassphrase)
+}
+
+func (mw *MultiWallet) CreateWatchOnlyWallet(walletName string, extendedPublicKey string) (*LibWallet, error) {
+
+	exists, err := mw.WalletNameExists(walletName)
+	if err != nil {
+		return nil, err
+	} else if exists {
+		return nil, errors.New(ErrWalletNotLoaded)
+	}
+
+	w := &Wallet{
+		Name:               walletName,
+		DiscoveredAccounts: true,
+	}
+
+	err = mw.db.Save(w)
```

Comment:
Add a comment here explaining that we're saving this db now - so we can get an autogenerated ID.